web/src/overlays.ts

```diff
@@ -301,28 +320,28 @@ export function close_modal(selector, conf = {}) {
     Micromodal.close(selector);
 }
 
-export function close_active_modal() {
+export function close_active_modal(): void {
     if (!is_modal_open()) {
         blueslip.warn("close_active_modal() called without checking is_modal_open()");
         return;
     }
 
     const $micromodal = $(".micromodal.modal--open");
-    Micromodal.close(`${CSS.escape($micromodal.attr("id"))}`);
+    Micromodal.close(`${CSS.escape($micromodal.attr("id") || "")}`);
 }
 
-export function close_for_hash_change() {
+export function close_for_hash_change(): void {
     $("div.overlay.show").removeClass("show");
-    if ($active_overlay) {
+    if ($active_overlay && close_handler) {
         close_handler();
     }
 }
 
-export function initialize() {
+export function initialize(): void {
     $("body").on("click", "div.overlay, div.overlay .exit", (e) => {
         let $target = $(e.target);
 
-        if (document.getSelection().type === "Range") {
+        if (document.getSelection()?.type === "Range") {
```

Comment:
Is this `?` necessary? When does https://developer.mozilla.org/en-US/docs/Web/API/Document/getSelection return undefined?

Reply:
@timabbott We will have to set `"strictNullChecks": false` in our tsconfig to remove that `?`, because apparently when `strictNullChecks` is enabled `document.getSelection()` return type becomes `Selection | null` 😕

Reply:
OK I think the answer is that this can be null in some unknown situation, and the code is correct.

https://microsoft.github.io/PowerBI-JavaScript/modules/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.html#getselection